src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/NewAzureStorageAccountEncryptionKeySource.cs

```diff
@@ -0,0 +1,92 @@
+{
+    /// <summary>
+    /// Create PSKeySource Object which will be used in Set-AzureRmStorageAccount as keysource
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, StorageAccountKeySourceStr, SupportsShouldProcess = false), OutputType(typeof(PSKeySource))]
+    public class NewAzureStorageEncryptionKeySourceCommand : StorageAccountBaseCmdlet
+    {
+
+        /// <summary>
+        /// Blob Pipeline parameter set name
+        /// </summary>
+        private const string StorageParameterSet = "StorageKeySourceInstance";
+
+        /// <summary>
+        /// Blob Pipeline parameter set name
+        /// </summary>
+        private const string KeyVaultParameterSet = "KeyVaultKeySourceInstance";
+
+        [Parameter(Position = 0, HelpMessage = "Storage Account encryption keySource Type", Mandatory = true, ParameterSetName = StorageParameterSet)]
+        [Parameter(Position = 0, HelpMessage = "Storage Account encryption keySource Type", Mandatory = true, ParameterSetName = KeyVaultParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PsKeySourceTypeEnum Type { get; set; }
+
+
+        [Parameter(HelpMessage = "Storage Account encryption keySource KeyVault KeyName",
+            Mandatory = true,
+            ParameterSetName = KeyVaultParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyName { get; set; }
+
+        [Parameter(HelpMessage = "Storage Account encryption keySource KeyVault KeyVersion",
+            Mandatory = true,
+            ParameterSetName = KeyVaultParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyVersion { get; set; }
+
+        [Parameter(HelpMessage = "Storage Account encryption keySource KeyVault KeyVaultUri",
+            Mandatory = true,
+            ParameterSetName = KeyVaultParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyVaultUri { get; set; }
+
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+
+            PSKeySource keySource = new PSKeySource();
+
+            if (ParameterSetName == StorageParameterSet)
+            {
+                if (Type != PsKeySourceTypeEnum.MicrosoftStorage)
+                    throw new System.ArgumentException("KeySource Type \"MicrosoftStorage\" must be choosed when not specify KeyName, KeyVersion, KeyVaultUri.");
+                keySource.KeySource = Type;
+            }
+            else
+            {
+                if (Type != PsKeySourceTypeEnum.MicrosoftKeyvault)
+                    throw new System.ArgumentException("KeySource Type \"MicrosoftKeyVault\" must be choosed when specify KeyName, KeyVersion, KeyVaultUri.");
+                keySource.KeySource = Type;
+                keySource.Keyvaultproperties.Keyname = KeyName;
+                keySource.Keyvaultproperties.KeyVersion = KeyVersion;
+                keySource.Keyvaultproperties.KeyVaultUri = KeyVaultUri;
+            }
+
+            this.WriteObject(keySource);
```

Comment:
The Key Vault or Key not include all necessary information, so we can't use a Key Vault encryption as Key source.